File: pyEpiabm/pyEpiabm/tests/test_property/test_infection_status.py
import unittest

from pyEpiabm.property import InfectionStatus


class TestInfectionStatus(unittest.TestCase):
    """Test the 'InfectionStatus' enum.
    """

    def test_construct(self):
        statuses = [ # noqa
            InfectionStatus.Susceptible,
            InfectionStatus.Exposed,
            InfectionStatus.InfectASympt,
            InfectionStatus.InfectMild,
            InfectionStatus.InfectGP,
            InfectionStatus.InfectHosp,
            InfectionStatus.InfectICU,
            InfectionStatus.InfectICURecov,
            InfectionStatus.Recovered,
            InfectionStatus.Dead]


if __name__ == '__main__':
    unittest.main()

File: cEpiabm/test/dataclasses/test_infection_status.cpp

#include "dataclasses/cell.hpp"

#include "../catch/catch.hpp"

#include <random>

using namespace epiabm;

TEST_CASE("dataclasses/infection_status: test status_string", "[InfectionStatus]")
{
    REQUIRE_NOTHROW(status_string(InfectionStatus::Susceptible));
    REQUIRE_NOTHROW(status_string(InfectionStatus::Exposed));
    REQUIRE_NOTHROW(status_string(InfectionStatus::InfectASympt));
    REQUIRE_NOTHROW(status_string(InfectionStatus::InfectMild));
    REQUIRE_NOTHROW(status_string(InfectionStatus::InfectGP));
    REQUIRE_NOTHROW(status_string(InfectionStatus::InfectHosp));
    REQUIRE_NOTHROW(status_string(InfectionStatus::InfectICU));
    REQUIRE_NOTHROW(status_string(InfectionStatus::InfectICURecov));
    REQUIRE_NOTHROW(status_string(InfectionStatus::Recovered));
    REQUIRE_NOTHROW(status_string(InfectionStatus::Dead));
    REQUIRE_NOTHROW(status_string(static_cast<InfectionStatus>(20)));
}
